openghg/standardise/surface/_icos_corso.py

```diff
@@ -0,0 +1,465 @@
+    species: str,
+    input_site: str,
+    site_fname: str,
+    input_inlet: str | None = None,
+    inlet_height_fname: str | None = None,
+) -> tuple[str, str, str | None]:
+    """
+    This function sets up the inital checks for site, inlet and standardises to internal format.
+
+    Args:
+        species: species_fname
+        input_site: user specified site name
+        site_fname: site name fetched from filename
+        input_inlet: user specified site name
+        inlet_height_fname: inlet fetched from filename
+
+    Returns:
+        site, input_site, inlet_height_magl
+    """
+
+    if input_site.lower() != site_fname.lower():
+        raise ValueError("Site mismatch between site argument passed and filename.")
+
+    inlet_height_fname = format_inlet(inlet_height_fname)
+    species = synonyms(species)
+
+    if inlet_height_fname is not None:
+        if input_inlet is not None and inlet_height_fname.lower() != input_inlet:
+            raise ValueError("Mismatch between inlet height passed and in filename.")
+
+    inlet_height_magl = format_inlet(inlet_height_fname, key_name="inlet_height_magl")
+
+    return species, input_site, inlet_height_magl
+
+
+def set_time_as_dataframe_index(dataframe: pd.DataFrame) -> pd.DataFrame:
+    """
+    Sets time as the dataframe index based on the dataset type
+    Args:
+        dataframe: Pandas dataframe
+
+    Returns:
+        dataframe: Pandas dataframe
+    """
+
+    dataframe["sampling_start"] = pd.to_datetime(dataframe["sampling_start"]).values.astype("datetime64[ns]")
+    dataframe["sampling_end"] = pd.to_datetime(dataframe["sampling_end"]).values.astype("datetime64[ns]")
+    dataframe["time"] = dataframe[["sampling_start", "sampling_end"]].min(axis=1)
+    dataframe["time"] = dataframe["time"].values.astype("datetime64[ns]")
+
+    dataframe.set_index("time", inplace=True)
+
+    return dataframe
+
+
+def calculate_sampling_period(dataframe: pd.DataFrame, species: str, header: list) -> pd.DataFrame:
+    """This function is used to calculate the difference between sampling_start and sampling_end to calculate the sampling period for each of the data points
+
+    Args:
+        df: Accepts pandas dataframe
+        species: species name
+        header: file header
+
+    returns: dataframe containing sampling_period column
+    """
+    columns = dataframe.columns
+    integration_time_flag = any("integrationtime is given in days" in s.lower() for s in header)
+
+    if not integration_time_flag:
+        if "sampling_end" in columns:
+            dataframe[f"{species}_sampling_period"] = (
+                dataframe["sampling_end"] - dataframe["sampling_start"]
+            ).dt.total_seconds()
+        else:
+            ValueError(
+                f"Unable to find sampling_start, sampling_end, or  {species}_sampling_period in the data to convert the time values into seconds"
+            )
+    else:
+        dataframe[f"{species}_sampling_period"] = pd.to_timedelta(
```

Comment:
If the column is named "integrationtime" or "integration_time" these lines can be updated to use this column to create the sampling period in seconds after the for this being in days:
```
    else:
        dataframe[f"{species}_sampling_period"] = pd.to_timedelta(
            dataframe["integration_time"], unit="D"
        ).dt.total_seconds()
```
The additional